packages/ui/src/components/Link/SLink.vue

```diff
@@ -0,0 +1,105 @@
+<script lang="ts">
+export default defineComponent({name: 'SLink'})
+</script>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import { usePropTypeFilter } from '@/composables/prop-type-filter'
+import { IconBasicExternalLink24, IconStatusInfo } from '../icons'
+import { LINK_ICON_POSITION_VALUES, LINK_TYPE_VALUES } from './consts'
+import { LinkIconPosition, LinkType } from './types'
+
+const props = withDefaults(
+    defineProps<{
+      type?: LinkType,
+      iconPosition?: LinkIconPosition,
+      icon?: boolean,
+      tag?: string | object
+    }>(),
+    {
+      type: 'link',
+      iconPosition: 'right',
+      icon: false,
+      tag: 'a'
+    },
+)
+
+
+const definitelyType = usePropTypeFilter(() => props.type, LINK_TYPE_VALUES, 'link')
+const definitelyIconPosition = usePropTypeFilter(() => props.iconPosition, LINK_ICON_POSITION_VALUES, 'right')
+const finalTag = computed(() => props.tag || 'a')
```

Comment:
Unnecessary computed

Reply:
I wanted to protect from emty string, but ok.